snowpack/src/config.ts

```diff
@@ -421,7 +421,14 @@ function normalizeConfig(_config: SnowpackUserConfig): SnowpackConfig {
     config.packageOptions.rollup.plugins = config.packageOptions.rollup.plugins || [];
   }
   config.exclude = Array.from(
-    new Set([...ALWAYS_EXCLUDE, `${config.buildOptions.out}/**`, ...config.exclude]),
+    new Set([
+      ...ALWAYS_EXCLUDE,
+      // Always ignore the final build directory.
+      `${config.buildOptions.out}/**`,
+      // If you've mounted your current working directory, ignore the node_modules directory.
```

Comment:
I could see this accidentally including node_modules in a multirepo setup (e.g. if we mount `./packages/foo`, we probably don't want `./packages/foo/node_modules` included).

Ideally, I think we'd still ignore `node_modules` but only the ones nested under the mount directory. Maybe something like this would do the trick:

```diff
    const files = (await new fdir()
      .withFullPaths()
      // Note: exclude() only matches directories, and not files. However, the cost
      // of false positives here is minor, so do this as a quick check to possibly
      // skip scanning into entire folder trees.
-     .exclude((_, dirPath) => foundExcludeMatch(dirPath))
+     .exclude((_, dirPath) => foundExcludeMatch(path.relative(mountKey, dirPath)))
      .crawl(mountKey)
      .withPromise()) as string[];
```

Reply:
ah right, we're probably getting this protection for free right now.

I really wanted to avoid making a change to every exclude call, but you're right that this is probably the most clear expression of what we're trying to accomplish here. Mount anything, but never travel down into a nested node_modules within a mount (since those are seen as "third-party" to whatever was just mounted).

Reply:
```
// if this is mounted:
/Fred/Code/mymonorepo/node_modules/astro/internal": "/internal"

/Fred/Code/myapp/node_modules/astro/internal/node_modules/...
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ should not be picked up by our automatic "node_modules" exclusion.
                                            ^^^^^^^^^^^^^^ anything inside of this directory should be ignored.

// "any node_modules that is explictly mounted should not be ignored. otherwise, ignore it."
```